### PR TITLE
Merge values to yaml

### PR DESCRIPTION
**Pull request by Tomasz-Smelcerz-SAP**

### CODE REVIEW

**Review by strekm — APPROVED** [review]

File: components/installer/pkg/overrides/generic-overrides.go [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+	}
+
+	err := yaml.Unmarshal([]byte(value), &target)
+	if err != nil {
+		return nil, err
+	}
+	return target, nil
+}
+
+//ToYaml converts OverridesMap to yaml
```
Comment (Tomasz-Smelcerz-SAP) [inline components/installer/pkg/overrides/generic-overrides.go]: yaml => YAML

File: components/installer/pkg/overrides/generic-overrides.go [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+	}
+
+	res, err := yaml.Marshal(oMap)
+	if err != nil {
+		return "", err
+	}
+	return string(res), nil
+}
+
+func FlattenMap(oMap OverridesMap) map[string]string {
```
Comment (Tomasz-Smelcerz-SAP) [inline components/installer/pkg/overrides/generic-overrides.go]: Add comment

File: components/installer/pkg/overrides/generic-overrides.go [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+		//Two maps case! Reccursion happens here!
+		MergeMaps(bMapVal, oMapVal)
+	} else {
+		//All other cases, even "pathological" one: When bMapVal is a Map but overrideVal is a string.
+		baseMap[key] = overrideVal
+	}
+}
+
+//MergeMaps merges all values from newOnes map into baseMap, overwriting final keys (string values) if both maps contain such entries
+func MergeMaps(baseMap, overridesMap OverridesMap) {
```
Comment (Tomasz-Smelcerz-SAP) [inline components/installer/pkg/overrides/generic-overrides.go]: Better description, move it towards top of the file

File: components/installer/pkg/overrides/generic-overrides.go [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+			mergeInto(baseMap, key, overrideVal)
+		} else {
+			//baseMap does not contain such entry. Just use overrideVal and we're done.
+			baseMap[key] = overrideVal
+		}
+	}
+}
+
+// Flattens given OverridesMap. The keys in result map will contain all intermediate keys joined with dots, e.g.: "istio.ingress.service.gateway: xyz"
+func _flattenMap(oMap OverridesMap, keys string, result map[string]string) {
```
Comment (Tomasz-Smelcerz-SAP) [inline components/installer/pkg/overrides/generic-overrides.go]: get rid of leading "_"

File: components/installer/pkg/overrides/generic-overrides.go [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+	for key, value := range sourceMap {
+		keys := strings.Split(key, ".")
+		mergeIntoMap(keys, value, mergedMap)
+	}
+
+	return mergedMap
+}
+
+//Helper function that contains merge logic
+func mergeInto(baseMap map[string]interface{}, key string, overrideVal interface{}) {
```
Comment (Tomasz-Smelcerz-SAP) [inline components/installer/pkg/overrides/generic-overrides.go]: Move it downwards

File: components/installer/pkg/overrides/generic-overrides.go (hunk 4 above)
Comment (Tomasz-Smelcerz-SAP) [reply components/installer/pkg/overrides/generic-overrides.go]: Do we even need this function at all?

File: components/installer/pkg/overrides/generic-overrides.go (hunk 1 above)
Comment (Tomasz-Smelcerz-SAP) [reply components/installer/pkg/overrides/generic-overrides.go]: Done.

File: components/installer/pkg/overrides/generic-overrides.go (hunk 2 above)
Comment (Tomasz-Smelcerz-SAP) [reply components/installer/pkg/overrides/generic-overrides.go]: Done.

File: components/installer/pkg/overrides/generic-overrides.go (hunk 3 above)
Comment (Tomasz-Smelcerz-SAP) [reply components/installer/pkg/overrides/generic-overrides.go]: Done.

File: components/installer/pkg/overrides/generic-overrides.go (hunk 4 above)
Comment (Tomasz-Smelcerz-SAP) [reply components/installer/pkg/overrides/generic-overrides.go]: Done. Function stays, will be removed later if unnecessary.

**Pull request closed without merging**